app/models/my_academics/my_academic_roles.rb

```diff
@@ -7,30 +7,38 @@ class MyAcademicRoles < UserSpecificModel
     def get_feed_internal
       {
         current: get_current_roles,
-        historical: get_historical_roles
+        historical: get_historical_roles,
       }
     end
 
     def get_current_roles
+      current_roles = role_defaults
+      map_roles(current_roles, current_term_cpp_roles)
+      map_roles(current_roles, student_group_roles)
+      current_roles
+    end
+
+    def student_group_roles
+      group_roles = []
+      student_group_codes.each do |group_code|
+        group_roles << Concerns::AcademicRoles.get_student_group_roles(group_code)
+      end
+      group_roles.flatten.uniq
+    end
+
+    def current_term_cpp
       current_term = Berkeley::Terms.fetch.current.try(:campus_solutions_id)
-      current_term_cpp = term_cpp.select {|t| t['term_id'].to_s >= current_term.to_s }
-      get_roles(current_term_cpp)
+      term_cpp.select {|t| t['term_id'].to_s >= current_term.to_s }
     end
 
-    def get_roles(terms)
+    def current_term_cpp_roles
```

Comment:
I have to dive into `EdoOracle::Queries` to figure out what `cpp` is. Can this be named something more like `current_term_career_program_and_plan_roles`? It's wordy, but much less obscure.